kernel/src/log_replay.rs

```diff
@@ -0,0 +1,182 @@
+//! This module provides structures and functionality to faciliate the log replay process.
```

Comment:
nit: not very useful, we should say exactly what's here, what we mean by 'log replay' etc.